### PR TITLE
Fix issue #34 and OTRS-1000047649

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

File: files/t/qsub-commands.t [hunk 1]
```diff
@@ -323,4 +323,15 @@ my $txt3 = "srun";
 ok(index($cmdstr, $txt) < index($cmdstr, $txt2), "$txt should be before $txt2 in \"$cmdstr\"");
 ok(index($cmdstr, $txt2) < index($cmdstr, $txt3), "$txt2 should be before $txt3 in \"$cmdstr\"");
 
+local $ENV{VSC_HOME} = '/home/path';
+$txt = '/home/path/test.foo';
+$stdin = "#!/usr/bin/bash\n#PBS -o \${VSC_HOME}/test.foo \necho\n";
+($cmdstr, $newtxt) = pst($stdin);
+isnt(index($newtxt, $txt), -1, "If #PBS -o \${VSC_HOME}/test.foo used in the submit script, it have to be translated to \"$txt\" in: $newtxt");
```
Comment (boegel) [inline files/t/qsub-commands.t]: `it has to be`

File: files/t/qsub-commands.t [hunk 2]
```diff
@@ -323,4 +323,15 @@ my $txt3 = "srun";
...
+
+local $ENV{MAIL} = 'e@mail.com';
+$stdin = "#!/usr/bin/bash\n#PBS -M \$PBS_O_MAIL \necho\n";
+($cmdstr, $newtxt) = pst($stdin);
+isnt(index($newtxt, $ENV{MAIL}), -1, "If #PBS -M \$PBS_O_MAIL used in the submit script, it have to be translated to \"$ENV{MAIL}\" in: $newtxt");
```
Comment (boegel) [inline files/t/qsub-commands.t]: `it has to be`

File: files/qsub.pl [hunk 3]
```diff
@@ -540,7 +540,19 @@ sub parse_script
...
+            $line =~ s/\$\{?$prefix($env_var)\}?/$ENV{$1}/g;
+        }
+        return ($line);
+    };
+
     foreach my $line (@lines) {
+        $line = &$change_line($line, 'VSC[A-Z_]*', '');
+        $line = &$change_line($line, 'HOME', '');
+        $line = &$change_line($line, 'HOME', 'PBS_O_');
+        $line = &$change_line($line, 'MAIL', 'PBS_O_');
```
Comment (boegel) [inline files/qsub.pl]: Maybe also include `$PBS_O_WORKDIR` (which should be replaced with the value of `$PWD`)?

File: files/qsub.pl [hunk 4]
```diff
@@ -540,7 +540,19 @@ sub parse_script
     # replace PBS_JOBID in -o / -e
     #   torque sets stdout/err in cwd -> so force abspath like done above
     # All script changes here
+    my $change_line = sub {
```
Comment (boegel) [inline files/qsub.pl]: I would give this a more meaningful name, like `replace_env_vars`?

Also, please include a line or two of comments above to describe what it does, give an example?

File: files/qsub.pl [hunk 5]
```diff
@@ -540,7 +540,19 @@ sub parse_script
...
+        my ($line, $env_var, $prefix) = @_;
+        if ($line =~ m/^\s*#PBS.*?\$\{?$prefix($env_var)\}?/ && $ENV{$1}) {
```
Comment (boegel) [inline files/qsub.pl]: What if there are multiple environment variables on a single line `#PBS` line?

File: files/qsub.pl (hunk 3 above)
Comment (boegel) [reply files/qsub.pl]: Also, can we make this a list you iterate over calling the function?

File: files/qsub.pl (hunk 5 above)
Comment (hajgato) [reply files/qsub.pl]: @boegel I would like to see an example. (I will fix it anyway)

Comment (hajgato) [reply files/qsub.pl]: (and actually the current script changes all the env_vars (see the `g` modifier after the last `/`)

Comment (hajgato) [reply files/qsub.pl]: ^in the next line

Comment (boegel) [reply files/qsub.pl]: OK, then just add a test case for it. Example:

```
#PBS -o $VSC_DATA/$VSC_INSTITUTE_CLUSTER/results

**Comment by hajgato** [comment]
@boegel fixed.

File: files/qsub.pl [hunk 6]
```diff
@@ -537,10 +537,39 @@ sub parse_script
...
+    # "#PBS -m $PBS_O_MAIL" to "#PBS -m e@mail.to", if $MAIL has the value "e@mail.to" at submit time.
+    my $replace_env_var = sub {
+        my ($line, $script_var, $env_var) = @_;
+        if ($line =~ m/^\s*#PBS.*?\$\{?$script_var\}?/ && $ENV{$env_var}) {
+            $line =~ s/\$\{?$script_var\}?/$ENV{$env_var}/g;
+        }
+        return ($line);
+    };
+
+    my @replace_script_vars = (
```
Comment (stdweird) [inline files/qsub.pl]: why is this not a hash?

File: files/qsub.pl [hunk 7]
```diff
@@ -537,10 +537,39 @@ sub parse_script
...
+        [qw(PBS_O_HOME HOME)],
+        [qw(PBS_O_HOST HOST)],
+        [qw(PBS_O_LOGNAME LOGNAME)],
+        [qw(PBS_O_MAIL MAIL)],
+        [qw(PBS_O_PATH PATH)],
+        [qw(PBS_O_SHELL SHELL)],
+        [qw(PBS_O_WORKDIR PWD)],
+    );
+    #Add all local varibales to @replace_script_vars
+    foreach my $existing_env_vars (keys %ENV) {
```
Comment (stdweird) [inline files/qsub.pl]: always use `sort keys`, it's reproducible that way

**Pull request merged**